src/aggregator/aggregator/aggregator.go

```diff
@@ -394,8 +394,10 @@ func (agg *aggregator) shardFor(id id.RawID) (*aggregatorShard, error) {
 		shardID = agg.shardFn(id, uint32(numShards))
 	}
 
+	// Maintain the rlock as long as we're accessing agg.shards (since it can be mutated otherwise).
 	agg.RLock()
 	if int(shardID) >= len(agg.shards) {
+		agg.RUnlock()
 		return nil, errShardNotOwned
 	}
```

Comment:
can we just get rid of this? I think I kept everything as-is from before the PR, and it's def overkill.

Reply:
get rid of the check and just rely on the map lookup, this should _never_ happen

Reply:
FYI `agg.shards` is a slice, and not a map (unless you had something else in mind @vdarulis )

Reply:
🤦